src/transformers/feature_extraction_utils.py

```diff
@@ -382,6 +381,22 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike], push_to_hub:
             kwargs (`Dict[str, Any]`, *optional*):
                 Additional key word arguments passed along to the [`~utils.PushToHubMixin.push_to_hub`] method.
         """
+        token = kwargs.pop("token", None)
+        use_auth_token = kwargs.pop("use_auth_token", None)
+
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
+            )
+            if token is not None:
+                raise ValueError(
+                    "`token` and `use_auth_token` are both specified. Please set only the argument `token`."
+                )
+            token = use_auth_token
+
+        if token is not None:
+            kwargs["token"] = token
```

Comment:
If `use_auth_token` is not in the argument (explicitly), and isn't popped up (on the `main)` branch, we need to add `kwargs["token"] = token` in this PR